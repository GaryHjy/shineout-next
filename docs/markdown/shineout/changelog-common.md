## 3.1.0-rc.1
2024-04-17

### 🆕 Feature

### 🐞 BugFix


### 💎 Enhancement

### 💅 Style

### 🆎 Type

### 🎨 Theme

- 调整 `setToken` 支持更新组件 token