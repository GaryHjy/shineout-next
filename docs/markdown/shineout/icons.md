`````
开发指南

# 自定义Icon

组件内的 Icon 支持替换为自定义 Icon
`````

## 用法

使用 setIcons 方法替换 Icon

```js
import { setIcons } from 'shineout'
setIcons({
  select: {
    DropdownArrow: <Icon name="angle-down" />,
  }
})
```


## 配置
支持修改的Icon
```js
const config = {
  alert: {
    Close: icons.Close,
    Info: icons.PcInfoCircleFill,
    Success: icons.PcCheckCircleFill,
    Warning: icons.PcWarningCircleFill,
    Danger: icons.PcWarningCircleFill,
    ConfirmWarning: icons.PcWarningCircleFill,
    Error: icons.PcWarningCircleFill,
    Confirm: icons.PcHelpCircleFill,
  },
  breadcrumb: {
    DropdownArrow: icons.ArrowDown,
  },
  card: {
    CollapseArrow: icons.ArrowRight,
  },
  carousel: {
    Backward: icons.ArrowLeft,
    Forward: icons.ArrowRight,
  },
  cascader: {
    More: icons.ArrowDown,
    DropdownArrow: icons.ArrowDown,
    Close: icons.PcCloseCircleFill,
    CollapseArrow: icons.ArrowRight,
  },
  collapse: {
    collapseArrow: icons.ArrowRight,
  },
  datepicker: {
    Close: icons.CloseOpaqueMultic1,
    Time: icons.Time,
    Calendar: icons.Calendar,
    ArrowDoubleLeft: icons.AngleDoubleLeft,
    ArrowLeft: icons.ArrowLeft,
    ArrowDoubleRight: icons.AngleDoubleRight,
    ArrowRight: icons.ArrowRight,
  },
  dropdown: {
    DropdownArrow: icons.ArrowDown,
  },
  editableArea: {
    Close: icons.CloseOpaqueMultic1,
  },
  image: {
    Close: icons.CloseOpaqueMultic1,
    Pics: icons.Pics2,
    Download: icons.Download,
    Preview: icons.Preview,
    LoadFail: icons.LoadingError2,
  },
  input: {
    ArrowRight: icons.ArrowRight,
    ArrowLeft: icons.ArrowLeft,
    Hide: icons.Hide,
    Show: icons.Display,
    Close: icons.CloseOpaqueMultic1,
  },
  menu: {
    CollapseArrow: icons.ArrowDown,
    FrontSolidArrowDown: icons.PcArrowFillDown,
  },
  modal: {
    Close: Icons?.Close,
  },
  pagination: {
    PrePage: icons.ArrowLeftDouble,
    NetPage: icons.ArrowRightDouble,
    More: icons.More,
    NextInButton: icons.ArrowRight,
    PreInButton: icons.ArrowLeft,
  },
  progress: {
    InfoCircle: icons.Info,
    WarningCircle: icons.Warning,
    SuccessCircle: icons.Check,
    DangerCircle: icons.Close,
    InfoLine: icons.PcInfoCircleFill,
    WarningLine: icons.PcWarningCircleFill,
    SuccessLine: icons.PcCheckCircleFill,
    DangerLine: icons.PcCloseCircleFill,
  },
  rate: {
    Star: icons.PcStarFill,
  },
  select: {
    Check: icons.Check,
    More: icons.More,
    DropdownArrow: icons.ArrowDown,
    Close: icons.CloseOpaqueMultic1,
  },
  steps: {
    Finish: icons.Check,
    Error: icons.Close,
  },
  table: {
    SortUp: icons.TableSortIconUp,
    SortDown: icons.TableSortIconDown,
    Expand: icons.Expand,
    Collapse: icons.OdecShrink,
  },
  tabs: {
    Pre: icons.ArrowLeft,
    Next: icons.ArrowRight,
    CollapseArrow: icons.ArrowLeft,
  },
  transfer: {
    DeleteAll: icons.Delete,
    DeleteItem: icons.Close,
    Search: icons.Search,
    Add: icons.ArrowRight,
    Remove: icons.ArrowLeft,
  },
  tree: {
    LineExpand: icons.TreeMinus,
    LineCollapse: icons.TreePlus,
    Expand: icons.TreeArrow,
  },
  treeSelect: {
    More: icons.More,
    DropdownArrow: icons.ArrowDown,
    Close: icons.CloseOpaqueMultic1,
  },
  upload: {
    File: icons.File,
    Success: icons.PcCheckCircleFill,
    Warning: icons.PcWarningCircleFill,
    Recover: icons.Return,
    Delete: icons.Delete,
    DeleteImage: icons.CloseOpaqueMultic1,
    RecoverImage: icons.UndeleteOpaque,
    PreviewImage: icons.Preview,
    AddImage: icons.Add,
  },
}
```

<!--
```react
import React from 'react';
import { icons } from 'shineout';
function MyIcon(props: any){
  const style={
    width: 20,
    height: 20,
    display: 'flex',
    alignItems: 'center',
    margin: '0 auto',
    color: 'var(--soui-input-icon-color,var(--soui-neutral-text-4,#666C7C))',
    ...props.style
  }
    return <div style={style}>{props.children}</div>;
}

type IconType = keyof typeof icons;

const App: React.FC = () => (
  <div>
    <h4>展示全部icons图标</h4>
    {
        Object.keys(icons).map((key) => {
            return <div key={key} style={{display: 'inline-block', width: 160, height: 100, textAlign: 'center'}}><MyIcon>{icons[key as IconType]}</MyIcon><div>{key}</div></div>
        })
    }
    </div>
);
export default App;
``` -->
