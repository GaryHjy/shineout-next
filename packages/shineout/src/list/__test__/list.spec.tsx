import React from 'react';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { List, TYPE } from 'shineout';
import mountTest from '../../tests/mountTest';
import {
  attributesTest,
  classTest,
  createClassName,
  displayTest,
  styleTest,
  textContentTest,
} from '../../tests/utils';
import { classLengthTest } from '../../tests/structureTest';

const SO_PREFIX = 'list';
const originClasses = ['wrapper', 'scrollContainer', 'row', 'item', 'empty'];
const originItemClasses = ['wrapperBordered', 'wrapperLarge', 'wrapperSmall', 'wrapperStriped'];
const {
  wrapper,
  scrollContainer,
  row,
  item: itemClassName,
  wrapperBordered,
  wrapperLarge,
  wrapperSmall,
  empty: emptyClassName,
  wrapperStriped,
} = createClassName(SO_PREFIX, originClasses, originItemClasses);

const defaultStyle = 'width: 100%;';

interface ListItem {
  id: number;
  lastName: string;
  firstName: string;
}

type ListProps = TYPE.List.Props<ListItem, ListItem>;
type ListData = ListProps['data'];
type ListRenderItem = ListProps['renderItem'];

const createMoreData = (num: number) => {
  const result: ListData = [];
  for (let i = 1; i <= num; i++) {
    result.push({
      id: i,
      firstName: 'test',
      lastName: `${i}`,
    });
  }
  return result;
};

const data: ListData = createMoreData(4);

const renderItem: ListRenderItem = (rowData) => (
  <div>{`Name: ${rowData.firstName}-${rowData.lastName}`}</div>
);

const RenderList = (props: any) => (
  <List data={data} keygen='id' renderItem={renderItem} {...props} />
);

afterEach(cleanup);
mountTest(<RenderList />);

describe('List[Base]', () => {
  displayTest(List as React.FC<any>, 'ShineoutList');
  test('should render when set className and style', () => {
    const style = { backgroundColor: 'red' };
    const sytleRender = 'background-color: red;';
    const className = 'test';
    const { container } = render(<RenderList style={style} className={className} />);
    const list = container.querySelector(wrapper)!;
    classTest(list, className);
    styleTest(list, sytleRender);
  });
  test('should render default', () => {
    const { container } = render(<RenderList />);
    const list = container.querySelector(wrapper)!;
    const listScrollContainer = list.querySelector(scrollContainer)!;
    const listRow = listScrollContainer.querySelectorAll(row);
    expect(listRow.length).toBe(data.length);
    listRow.forEach((i, index) => {
      const item = i.querySelector(itemClassName)!;
      styleTest(item, defaultStyle);
      textContentTest(item, `Name: ${data[index].firstName}-${data[index].lastName}`);
    });
  });
  test('should render without renderItem', () => {
    const dataWithoutRenderItem = ['red', 'orange', 'yellow', 'green', 'cyan', 'blue', 'violet'];
    const { container } = render(<List data={dataWithoutRenderItem} keygen />);
    const list = container.querySelector(wrapper)!;
    const rows = list.querySelectorAll(row);
    rows.forEach((item, index) => {
      textContentTest(item, dataWithoutRenderItem[index]);
    });
  });
  test('should render when set keygen is true with data item is object', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
    render(<RenderList keygen />);
    expect(
      JSON.stringify(errorSpy.mock.calls).indexOf(
        'Error: keygen result expect a string or a number, get',
      ),
    ).toBeTruthy();
  });
  test('should render when set bordered', () => {
    const { container } = render(<RenderList bordered />);
    const list = container.querySelector(wrapper)!;
    classTest(list, wrapperBordered);
  });
  test('should render when set size is large', () => {
    const { container } = render(<RenderList size='large' />);
    const list = container.querySelector(wrapper)!;
    classTest(list, wrapperLarge);
  });
  test('should render when set size is small', () => {
    const { container } = render(<RenderList size='small' />);
    const list = container.querySelector(wrapper)!;
    classTest(list, wrapperSmall);
  });
  test('should render when set height', () => {
    const height = 300;
    const { container } = render(<RenderList height={height} />);
    const list = container.querySelector(wrapper)!;
    styleTest(list, `height: ${height}px;`);
  });
  test('should render when set data is empty', () => {
    const emptyData: ListData = [];
    const { container } = render(<RenderList data={emptyData} />);
    const list = container.querySelector(wrapper)!;
    classLengthTest(list, emptyClassName, 1);
    classLengthTest(list.querySelector(emptyClassName)!, 'svg', 1);
  });
  test('should render when set empty', () => {
    const emptyContainer = 'none';
    const emptyData: ListData = [];
    const { container } = render(<RenderList data={emptyData} empty={emptyContainer} />);
    const listEmpty = container.querySelector(emptyClassName)!;
    textContentTest(listEmpty, emptyContainer);
  });
  test('should render when set striped', () => {
    const { container } = render(<RenderList striped />);
    const list = container.querySelector(wrapper)!;
    classTest(list, wrapperStriped);
  });
});
describe('List[Fixed]', () => {
  const virtualData = createMoreData(30);
  test('should render when set fixed', () => {
    const { container } = render(<RenderList data={virtualData} fixed />);
    const list = container.querySelector(wrapper)!;
    const listScroll = list.firstElementChild?.firstElementChild as Element;
    const listScrollContainer = listScroll.firstElementChild as Element;
    const listScrollMain = listScrollContainer.firstElementChild as Element;
    attributesTest(listScroll, 'data-soui-type', 'scroll');
    attributesTest(listScrollContainer, 'data-soui-type', 'scroll-container');
    styleTest(listScrollMain, 'transform: translate3d(0, -0px, 0);');
    fireEvent.scroll(listScroll, { target: { scrollTop: 50 } });
    screen.debug();
  });
});
