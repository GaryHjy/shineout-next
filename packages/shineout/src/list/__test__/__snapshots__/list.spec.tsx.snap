// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List[Base] should render correctly  1`] = `
<div
  class="c3wrw8q c51nw85"
>
  <div
    class="cv8iaaj"
  >
    <div
      class="cmhpght"
    >
      <div
        class="c-d2czxw"
        style="width: 100%;"
      >
        <div
          style="display: flex; align-items: center; width: 100%; font-size: 14px;"
        >
          <svg
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z"
              fill="#E8EBF0"
            />
            <path
              clip-rule="evenodd"
              d="M19.8301 21.0204C22.46 21.0204 24.592 18.8884 24.592 16.2585C24.592 13.6286 22.46 11.4966 19.8301 11.4966C17.2002 11.4966 15.0682 13.6286 15.0682 16.2585C15.0682 18.8884 17.2002 21.0204 19.8301 21.0204ZM16.7628 22.7211C13.9482 22.7211 11.6665 25.0028 11.6665 27.8174C11.6665 28.1963 11.9737 28.5034 12.3525 28.5034H27.6471C28.026 28.5034 28.3332 28.1963 28.3332 27.8174C28.3332 25.0028 26.0515 22.7211 23.2369 22.7211H16.7628Z"
              fill="#B3B7C1"
              fill-rule="evenodd"
            />
          </svg>
          <div
            style="flex: 1; min-width: 0; margin: 0px 12px;"
          >
            <div
              style="font-weight: bold;"
            >
              List Title
            </div>
            <div>
              Marketing Designer
            </div>
          </div>
          <div>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Preview
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Edit
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Delete
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="cmhpght"
    >
      <div
        class="c-d2czxw"
        style="width: 100%;"
      >
        <div
          style="display: flex; align-items: center; width: 100%; font-size: 14px;"
        >
          <svg
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z"
              fill="#E8EBF0"
            />
            <path
              clip-rule="evenodd"
              d="M19.8301 21.0204C22.46 21.0204 24.592 18.8884 24.592 16.2585C24.592 13.6286 22.46 11.4966 19.8301 11.4966C17.2002 11.4966 15.0682 13.6286 15.0682 16.2585C15.0682 18.8884 17.2002 21.0204 19.8301 21.0204ZM16.7628 22.7211C13.9482 22.7211 11.6665 25.0028 11.6665 27.8174C11.6665 28.1963 11.9737 28.5034 12.3525 28.5034H27.6471C28.026 28.5034 28.3332 28.1963 28.3332 27.8174C28.3332 25.0028 26.0515 22.7211 23.2369 22.7211H16.7628Z"
              fill="#B3B7C1"
              fill-rule="evenodd"
            />
          </svg>
          <div
            style="flex: 1; min-width: 0; margin: 0px 12px;"
          >
            <div
              style="font-weight: bold;"
            >
              List Title
            </div>
            <div>
              Financial Controller
            </div>
          </div>
          <div>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Preview
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Edit
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Delete
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="cmhpght"
    >
      <div
        class="c-d2czxw"
        style="width: 100%;"
      >
        <div
          style="display: flex; align-items: center; width: 100%; font-size: 14px;"
        >
          <svg
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z"
              fill="#E8EBF0"
            />
            <path
              clip-rule="evenodd"
              d="M19.8301 21.0204C22.46 21.0204 24.592 18.8884 24.592 16.2585C24.592 13.6286 22.46 11.4966 19.8301 11.4966C17.2002 11.4966 15.0682 13.6286 15.0682 16.2585C15.0682 18.8884 17.2002 21.0204 19.8301 21.0204ZM16.7628 22.7211C13.9482 22.7211 11.6665 25.0028 11.6665 27.8174C11.6665 28.1963 11.9737 28.5034 12.3525 28.5034H27.6471C28.026 28.5034 28.3332 28.1963 28.3332 27.8174C28.3332 25.0028 26.0515 22.7211 23.2369 22.7211H16.7628Z"
              fill="#B3B7C1"
              fill-rule="evenodd"
            />
          </svg>
          <div
            style="flex: 1; min-width: 0; margin: 0px 12px;"
          >
            <div
              style="font-weight: bold;"
            >
              List Title
            </div>
            <div>
              Development Lead
            </div>
          </div>
          <div>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Preview
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Edit
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Delete
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="cmhpght"
    >
      <div
        class="c-d2czxw"
        style="width: 100%;"
      >
        <div
          style="display: flex; align-items: center; width: 100%; font-size: 14px;"
        >
          <svg
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z"
              fill="#E8EBF0"
            />
            <path
              clip-rule="evenodd"
              d="M19.8301 21.0204C22.46 21.0204 24.592 18.8884 24.592 16.2585C24.592 13.6286 22.46 11.4966 19.8301 11.4966C17.2002 11.4966 15.0682 13.6286 15.0682 16.2585C15.0682 18.8884 17.2002 21.0204 19.8301 21.0204ZM16.7628 22.7211C13.9482 22.7211 11.6665 25.0028 11.6665 27.8174C11.6665 28.1963 11.9737 28.5034 12.3525 28.5034H27.6471C28.026 28.5034 28.3332 28.1963 28.3332 27.8174C28.3332 25.0028 26.0515 22.7211 23.2369 22.7211H16.7628Z"
              fill="#B3B7C1"
              fill-rule="evenodd"
            />
          </svg>
          <div
            style="flex: 1; min-width: 0; margin: 0px 12px;"
          >
            <div
              style="font-weight: bold;"
            >
              List Title
            </div>
            <div>
              Developer
            </div>
          </div>
          <div>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Preview
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Edit
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Delete
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`List[Base] should render correctly about border 1`] = `
<div
  style="padding: 0px 16px; border: 1px solid #e8ebf0;"
>
  <div
    class="c3wrw8q"
  >
    <div
      class="cv8iaaj"
    >
      <div
        class="cmhpght"
      >
        <div
          class="c-d2czxw"
          style="width: 100%; padding-left: 0px; padding-right: 0px;"
        >
          <div>
            Name: Ephraim-Wisozk
          </div>
        </div>
      </div>
      <div
        class="cmhpght"
      >
        <div
          class="c-d2czxw"
          style="width: 100%; padding-left: 0px; padding-right: 0px;"
        >
          <div>
            Name: Osvaldo-Beer
          </div>
        </div>
      </div>
      <div
        class="cmhpght"
      >
        <div
          class="c-d2czxw"
          style="width: 100%; padding-left: 0px; padding-right: 0px;"
        >
          <div>
            Name: Dylan-Ratke
          </div>
        </div>
      </div>
      <div
        class="cmhpght"
      >
        <div
          class="c-d2czxw"
          style="width: 100%; padding-left: 0px; padding-right: 0px;"
        >
          <div>
            Name: Shaniya-Jacobs
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`List[Base] should render correctly about colNum 1`] = `
<div
  class="hello c3wrw8q c51nw85"
  style="color: blue;"
>
  <div
    class="cv8iaaj"
  >
    <div
      class="cmhpght"
    >
      <div
        class="c-d2czxw"
        style="width: 50%;"
      >
        <div
          style="display: flex; align-items: center; width: 100%; font-size: 14px;"
        >
          <svg
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z"
              fill="#E8EBF0"
            />
            <path
              clip-rule="evenodd"
              d="M19.8301 21.0204C22.46 21.0204 24.592 18.8884 24.592 16.2585C24.592 13.6286 22.46 11.4966 19.8301 11.4966C17.2002 11.4966 15.0682 13.6286 15.0682 16.2585C15.0682 18.8884 17.2002 21.0204 19.8301 21.0204ZM16.7628 22.7211C13.9482 22.7211 11.6665 25.0028 11.6665 27.8174C11.6665 28.1963 11.9737 28.5034 12.3525 28.5034H27.6471C28.026 28.5034 28.3332 28.1963 28.3332 27.8174C28.3332 25.0028 26.0515 22.7211 23.2369 22.7211H16.7628Z"
              fill="#B3B7C1"
              fill-rule="evenodd"
            />
          </svg>
          <div
            style="flex: 1; min-width: 0; margin: 0px 12px;"
          >
            <div
              style="font-weight: bold;"
            >
              List Title
            </div>
            <div>
              Marketing Designer
            </div>
          </div>
          <div>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Preview
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Edit
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Delete
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="c-d2czxw"
        style="width: 50%;"
      >
        <div
          style="display: flex; align-items: center; width: 100%; font-size: 14px;"
        >
          <svg
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z"
              fill="#E8EBF0"
            />
            <path
              clip-rule="evenodd"
              d="M19.8301 21.0204C22.46 21.0204 24.592 18.8884 24.592 16.2585C24.592 13.6286 22.46 11.4966 19.8301 11.4966C17.2002 11.4966 15.0682 13.6286 15.0682 16.2585C15.0682 18.8884 17.2002 21.0204 19.8301 21.0204ZM16.7628 22.7211C13.9482 22.7211 11.6665 25.0028 11.6665 27.8174C11.6665 28.1963 11.9737 28.5034 12.3525 28.5034H27.6471C28.026 28.5034 28.3332 28.1963 28.3332 27.8174C28.3332 25.0028 26.0515 22.7211 23.2369 22.7211H16.7628Z"
              fill="#B3B7C1"
              fill-rule="evenodd"
            />
          </svg>
          <div
            style="flex: 1; min-width: 0; margin: 0px 12px;"
          >
            <div
              style="font-weight: bold;"
            >
              List Title
            </div>
            <div>
              Financial Controller
            </div>
          </div>
          <div>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Preview
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Edit
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Delete
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="cmhpght"
    >
      <div
        class="c-d2czxw"
        style="width: 50%;"
      >
        <div
          style="display: flex; align-items: center; width: 100%; font-size: 14px;"
        >
          <svg
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z"
              fill="#E8EBF0"
            />
            <path
              clip-rule="evenodd"
              d="M19.8301 21.0204C22.46 21.0204 24.592 18.8884 24.592 16.2585C24.592 13.6286 22.46 11.4966 19.8301 11.4966C17.2002 11.4966 15.0682 13.6286 15.0682 16.2585C15.0682 18.8884 17.2002 21.0204 19.8301 21.0204ZM16.7628 22.7211C13.9482 22.7211 11.6665 25.0028 11.6665 27.8174C11.6665 28.1963 11.9737 28.5034 12.3525 28.5034H27.6471C28.026 28.5034 28.3332 28.1963 28.3332 27.8174C28.3332 25.0028 26.0515 22.7211 23.2369 22.7211H16.7628Z"
              fill="#B3B7C1"
              fill-rule="evenodd"
            />
          </svg>
          <div
            style="flex: 1; min-width: 0; margin: 0px 12px;"
          >
            <div
              style="font-weight: bold;"
            >
              List Title
            </div>
            <div>
              Development Lead
            </div>
          </div>
          <div>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Preview
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Edit
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Delete
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="c-d2czxw"
        style="width: 50%;"
      >
        <div
          style="display: flex; align-items: center; width: 100%; font-size: 14px;"
        >
          <svg
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z"
              fill="#E8EBF0"
            />
            <path
              clip-rule="evenodd"
              d="M19.8301 21.0204C22.46 21.0204 24.592 18.8884 24.592 16.2585C24.592 13.6286 22.46 11.4966 19.8301 11.4966C17.2002 11.4966 15.0682 13.6286 15.0682 16.2585C15.0682 18.8884 17.2002 21.0204 19.8301 21.0204ZM16.7628 22.7211C13.9482 22.7211 11.6665 25.0028 11.6665 27.8174C11.6665 28.1963 11.9737 28.5034 12.3525 28.5034H27.6471C28.026 28.5034 28.3332 28.1963 28.3332 27.8174C28.3332 25.0028 26.0515 22.7211 23.2369 22.7211H16.7628Z"
              fill="#B3B7C1"
              fill-rule="evenodd"
            />
          </svg>
          <div
            style="flex: 1; min-width: 0; margin: 0px 12px;"
          >
            <div
              style="font-weight: bold;"
            >
              List Title
            </div>
            <div>
              Developer
            </div>
          </div>
          <div>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Preview
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Edit
              </span>
            </button>
            <button
              class="c-trc8a3 cvey2f3 c-nzauiu"
              type="button"
            >
              <span
                class="cboe6wq"
              >
                Delete
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`List[Base] should render correctly about empty 1`] = `
<div
  class="c3wrw8q c51nw85"
>
  <div
    class="cv8iaaj"
  />
  <div
    class="cl35yt0"
  >
    no data
  </div>
</div>
`;

exports[`List[Base] should render correctly about size 1`] = `
<div
  class="cw6sirj"
  style="margin-bottom: 24px;"
>
  <div
    class="c-emivhp"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="radio"
    />
    <span
      class="c-yzux30"
    >
      <span
        class="c-2ldwtt"
        tabindex="1"
      />
    </span>
    <span
      class="c-qgj373"
    >
      small
    </span>
  </div>
  <div
    class="c-emivhp c-wsxxws"
  >
    <input
      checked=""
      style="display: none;"
      tabindex="0"
      type="radio"
    />
    <span
      class="c-yzux30"
    >
      <span
        class="c-2ldwtt"
        tabindex="1"
      />
    </span>
    <span
      class="c-qgj373"
    >
      default
    </span>
  </div>
  <div
    class="c-emivhp"
  >
    <input
      style="display: none;"
      tabindex="0"
      type="radio"
    />
    <span
      class="c-yzux30"
    >
      <span
        class="c-2ldwtt"
        tabindex="1"
      />
    </span>
    <span
      class="c-qgj373"
    >
      large
    </span>
  </div>
</div>
`;

exports[`List[Base] should render correctly about striped 1`] = `
<div
  class="c3wrw8q c51nw85 c-dqmi77"
>
  <div
    class="cv8iaaj"
  >
    <div
      class="cmhpght"
    >
      <div
        class="c-d2czxw"
        style="width: 100%;"
      >
        <div>
          Name: Ephraim-Wisozk
        </div>
      </div>
    </div>
    <div
      class="cmhpght"
    >
      <div
        class="c-d2czxw"
        style="width: 100%;"
      >
        <div>
          Name: Osvaldo-Beer
        </div>
      </div>
    </div>
    <div
      class="cmhpght"
    >
      <div
        class="c-d2czxw"
        style="width: 100%;"
      >
        <div>
          Name: Dylan-Ratke
        </div>
      </div>
    </div>
    <div
      class="cmhpght"
    >
      <div
        class="c-d2czxw"
        style="width: 100%;"
      >
        <div>
          Name: Shaniya-Jacobs
        </div>
      </div>
    </div>
  </div>
</div>
`;
